### ISSUE TITLE
Node Download Times out causing App deploy failure

### ISSUE THREAD

**Issue by schneems** [heroku/heroku-buildpack-ruby#335]
Since moving from S3 to the new node endpoint the timeout isn't long enough sometimes. We're seeing ~100 deploys fail a day due to this. Ticket 208962

```
 !     Command: 'set -o pipefail; curl --fail --retry 3 --retry-delay 1 --connect-timeout 3 --max-time 30 http://nodejs.org/dist/v0.10.30/node-v0.10.30-linux-x64.tar.gz -s -o - | tar zxf - node-v0.10.30-linux-x64/bin/node' failed unexpectedly:
 !     
 !     gzip: stdin: invalid compressed data--format violated
 !     tar: Unexpected EOF in archive
 !     tar: Unexpected EOF in archive
 !     tar: Error is not recoverable: exiting now
```

We can bump the timeout or cache the node binary. The second is much harder as we need to also code invalidation logic, but it will make deploys faster.

**Comment by josegonzalez** [comment]
Was there a reason the deploys moved from s3 to the node endpoint? I assume cost, correct?

**Comment by schneems** [comment]
S3pository is on S3. This issue is closed by #336

**Comment by hone** [comment]
@josegonzalez we actually moved from the official node binary hosts to our own hosted/cached versions of it b/c the nodejs.org infrastructure seems to be unreliable more often than we'd like.
